test/test_log.py

```diff
@@ -162,21 +164,34 @@ def test_5(self):
         self.assertTrue(name_filter(record))
         self.assertEqual(record.name, '__init__/module')
 
-    def test_plugin_path_long(self):
+    def test_plugin_path_long_1(self):
         DebugOpt.PLUGIN_FULLPATH.enabled = True
-        record = FakeRecord(name=None, pathname='/path1/path2/plugins/path3/plugins/plugin.zip')
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip')
         self.assertTrue(name_filter(record))
-        self.assertEqual(record.name, 'plugins/path3/plugins/plugin')
+        self.assertEqual(record.name, 'user/picard/plugins/plugin')
+
+    def test_plugin_path_long_2(self):
+        DebugOpt.PLUGIN_FULLPATH.enabled = True
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip/plugin')
+        self.assertTrue(name_filter(record))
+        self.assertEqual(record.name, 'user/picard/plugins/plugin.zip/plugin')
```

Comment:
Why are we ending with a path not starting with `/` here?

Reply:
I believe that's because of stripping out the anchor in the lines in `log.py`:

```python
    if parts[0] == path.anchor:
        del parts[0]
```

That's the same behavior we're seeing when the path isn't relative to the `picard` module.  For example, in `test_log.py`:

```python
@unittest.skipIf(IS_WIN, "Posix test")
@patch('picard.log.picard_module_path', PurePosixPath('/picard'))
class NameFilterTestAbs(PicardTestCase):


    def test_1(self):
        record = FakeRecord(name=None, pathname='/path/module/file.py')
        self.assertTrue(name_filter(record))
        self.assertEqual(record.name, 'path/module/file')
```

If you want the absolute (non-shortened) paths to start with a "/" (for both Windows and POSIX) the code could be changed to:

```python
    if parts[0] == path.anchor:
        parts[0] = '/'
```

I'll make the change, because it will make it clearer that the path is absolute.